gradlew

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env sh
```

Comment:
do we need all these gradle files?  can't they be pulled in via some submodule?

Reply:
strictly speaking they are not necessary, but this a gradle bootstrapping code that makes life of a developer much easier. E.g. without this one I'm not able to build this project until and unless I somehow manage to create it (in my case, I'm using IntelliJ Idea to generate it for me; it also simplifies the readme steps: simply run the wrapper, otherwise it will include references to gradle docs and may be tricky for someone who is not developing Java code but wants to run the Java tests).